pkg/cert-watcher/watcher.go

```diff
@@ -0,0 +1,155 @@
+package certwatcher
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/k8tz/k8tz/pkg/version"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+var (
+	verboseLogger *log.Logger
+	warningLogger *log.Logger
+	infoLogger    *log.Logger
+	errorLogger   *log.Logger
+)
+
+type Watcher struct {
+	TLSCertFile     string
+	TLSKeyFile      string
+	SecretName      string
+	SecretNamespace string
+	Verbose         bool
+	clientset       kubernetes.Interface
+}
+
+func NewCertWatcher() *Watcher {
```

Comment:
we need to create test for CertWatcher. 

as far as i can see it will require to set a [fake](https://pkg.go.dev/k8s.io/client-go/kubernetes/fake) `clientset` and a cancelable context with timeout to `factory.Start()`. 
then we can then create a test suite that create/update a fake secret and watch the files created/updated on a [testing.T.TempDir()](https://pkg.go.dev/testing#T.TempDir).

Reply:
okay, will create the test case for this